base/shared/src/main/scala/scalaz/data/validation.scala

```diff
@@ -0,0 +1,113 @@
+package scalaz
+package data
+
+import scalaz.core.EqClass
+import scalaz.algebra.{ MonoidClass, SemigroupClass }
+import scalaz.ct._
+import scalaz.debug.DebugClass
+
+trait ValidationModule {
+  type Validation[A, B]
+
+  object Invalid {
+    def unapply[A, B](vab: Validation[A, B]): Option[A] = fold(vab)(Some(_), _ => None)
+  }
+
+  object Valid {
+    def unapply[A, B](vab: Validation[A, B]): Option[B] = fold(vab)(_ => None, Some(_))
+  }
+
+  /* functions */
+  def fold[A, B, C](vab: Validation[A, B])(ia: A => C, vb: B => C): C
+  def fromDisjunction[A, B](oa: A \/ B): Validation[A, B]
+  def invalid[L, R](value: L): Validation[L, R]
+  def valid[L, R](value: R): Validation[L, R]
+}
+
+private[data] object ValidationImpl extends ValidationModule {
+  type Validation[A, B] = A \/ B
+
+  final def fold[A, B, C](vab: Validation[A, B])(ia: A => C, vb: B => C): C = vab match {
+    case Invalid(a) => ia(a)
+    case Valid(b)   => vb(b)
+  }
+
+  @inline def fromDisjunction[L, R](disj: L \/ R): Validation[L, R] = disj
+
+  @inline def invalid[L, R](value: L): Validation[L, R] = -\/(value)
+  @inline def valid[L, R](value: R): Validation[L, R]   = \/-(value)
+
+  implicit def validationApplicative[L](implicit L: Semigroup[L]): Applicative[Validation[L, ?]] =
```

Comment:
I think these instances aren't going to get resolved where they are. You need a (`ValidationInstances` mixed into a) `ValidationModule` object, is what I think we're doing.

Reply:
Ah right. Forgot that part.  Will add it

Reply:
Looks like we're doing it a little differently for newts, but you were write about the discoverability problem